### PR TITLE
Audigent Segmentation Real-time Data Provider

### PR DESCRIPTION
**Pull request by anthonylauzon**
## Type of change
- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
This changeset adds a real time data module for retrieving Audigent user segmentation data for unified ids.  Using this real time data module, rtb adapters can supply rich segmentation data at request time in order to better target audiences.  

Audigent is a next-generation data management platform and a first-of-a-kind "data agency" containing some of the most exclusive content-consuming audiences across desktop, mobile and social platforms.

- ant@audigent.com
- [x] official adapter submission

### CODE REVIEW

File: modules/audigentRtdProvider.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  const userIds = (getGlobal()).getUserIds();
+  let tdid = null;
+
+  if (userIds && userIds['tdid']) {
+    tdid = userIds['tdid'];
+  } else {
+    onDone({});
+  }
+
+  const url = `//seg.ad.gt/api/v1/rtb_segments?tdid=${tdid}`;
```
Comment (bretg) [inline modules/audigentRtdProvider.js]: This has to be hardcoded to HTTPS - protocol relative URLs are no longer allowed.

File: integrationExamples/gpt/audigentSegments_example.html [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    <script>
+        googletag.cmd.push(function() { googletag.display('test-div'); });
+    </script>
+</div>
+TDID:
+<div id='tdid'>
+</div>
+
+Audigent Segments:
+<div id='audigent_segments'>
```
Comment (bretg) [inline integrationExamples/gpt/audigentSegments_example.html]: Don't understand why a div is needed.

**Comment by bretg** [comment]
@anthonylauzon - sorry, we're backed up. But just skimming this in a few seconds, I notice that you haven't provided unit tests. Please add them. We require 80% coverage.https://github.com/prebid/Prebid.js/tree/master/test/spec/modules

More importantly though, I don't understand what your module is actually doing. Note that I'm semi-technical, but don't have a lot of time to dig into this. Please explain how it's supposed to make the user segments available to bid adapters and why it needs a <div>.

File: integrationExamples/gpt/audigentSegments_example.html (hunk 2 above)
Comment (anthonylauzon) [reply integrationExamples/gpt/audigentSegments_example.html]: This is just here to show the resulting segments that could be attached to the bid request objects.

**Comment by anthonylauzon** [comment]
@brettg can you point me at a RTD module that has a unit test example?


This real-time data module provides first-party Audigent segments that can be attached to bid request objects destined for different SSPs in order to optimize targeting.  Audigent maintains a large database of first-party Tradedesk Unified ID to third party segment mappings that can now be queried at bid-time.

**Comment by bretg** [comment]
You can add tests to https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/realTimeModule_spec.js

**Comment by anthonylauzon** [comment]
Not sure why CircleCI is failing but 'gulp lint' is working on my machine

**Comment by anthonylauzon** [comment]
@bretg can you kick off another CircleCI build? I am not sure what these failures are resulting from.  `gulp test` passes locally.

**Comment by pm-harshad-mane** [comment]
Hello @anthonylauzon ,

I have triggered the CI job again, let's see.

File: modules/audigentRtdProvider.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  const url = `https://seg.ad.gt/api/v1/rtb_segments?tdid=${tdid}`;
+
+  ajax(url, {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          const data = JSON.parse(response);
+          setData(data);
+          if (data && data.audigent_segments) {
+            let dataToReturn = adUnits.reduce((rp, cau) => {
```
Comment (pm-harshad-mane) [inline modules/audigentRtdProvider.js]: you might want to set the data in local-storage only if it is valid.

**Comment by pm-harshad-mane** [comment]
Hello @anthonylauzon 
Could you please add documentation in "audigentRtdProvider.md" file about 
- how this sub-module is useful? 
- what data will be shared by the module? the format
- how bidders can use the shared data?
You might also want to add a usage example in https://github.com/prebid/Prebid.js/blob/master/modules/rtdModule/realTimeData.md

**Comment by anthonylauzon** [comment]
@pm-harshad-mane i've made the changes requested.  it's not clear to me why CircleCI is failing.  Are there non-deterministic tests in master?

**Comment by pm-harshad-mane** [comment]
@anthonylauzon the CI job is failing due to reasons unrelated to the changes you have made, I have restarted the CI job.

**Comment by anthonylauzon** [comment]
I think that the PR is ready for merge.  Are there any other modifications you would like made?

**Comment by pm-harshad-mane** [comment]
@anthonylauzon can you provide test credentials to try-out Audigent module?
I just wanted to see it in a working example.

**Comment by anthonylauzon** [comment]
@pm-harshad-mane no credentials needed.  just run `gulp server` and look at `http://localhost:9999/integrationExamples/gpt/audigentSegments_example.html`

**Comment by anthonylauzon** [comment]
Integration example added to the readme.

**Comment by jsnellbaker** [comment]
Hi @anthonylauzon 

I tried to load the test page you mentioned above with the noted modules, but it seems like I'm hitting a CORS error against the endpoint:
```
Access to XMLHttpRequest at 'https://seg.ad.gt/api/v1/rtb_segments?tdid=null' from origin 'http://localhost:9999' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

I noticed the `tdid` was `null` in this case.  Is there something extra I should do here?

**Comment by anthonylauzon** [comment]
@jsnellbaker you will need to use a localhost CORS allowing extension to allow the request to load or disable CORS for localhost in the browser.

**Pull request merged**